lib/unobtainium/drivers/appium.rb

```diff
@@ -108,35 +108,55 @@ def ensure_preconditions(_, _)
                 err.backtrace
         end
 
+        def option_set?(option)
+          option == "" or option.nil? ? false : true
+        end
+
         ##
         # Sanitize options, and expand the :browser key, if present.
         def resolve_options(label, options)
           # Normalize label and options
           normalized = normalize_label(label)
           options = ::Collapsium::UberHash.new(options || {})
 
-          # Merge 'caps' and 'desired_capabilities', letting the former win
-          options[:caps] =
-            ::Collapsium::UberHash.new(options['desired_capabilities'])
-                                  .recursive_merge(options[:desired_capabilities])
-                                  .recursive_merge(options[:caps])
-          options.delete(:desired_capabilities)
-          options.delete('desired_capabilities')
+          # if it's testdroid, we do not change and merge anything here
+          unless testdroid_testrun? options
+            # Merge 'caps' and 'desired_capabilities' into :caps, but leave the
+            # other one untouched
+            options[:caps] = ::Collapsium::UberHash.new(options['desired_capabilities'])
+                                                   .recursive_merge(options[:desired_capabilities])
+                                                   .recursive_merge(options[:caps])
+            options.delete(:desired_capabilities)
+            options.delete('desired_capabilities')
+          end
 
           # The label specifies the platform, if no other platform is given.
           if not options['caps.platformName']
             options['caps.platformName'] = normalized.to_s
           end
 
-          # Make the appium driver behave a little more like Selenium by using
-          # the :url key if the normalized label is remote, and setting
-          # appropriate options.
-          set_url = options['appium_lib.server_url']
-          if set_url and options['url'] and not set_url == options['url']
-            warn "You have the remote URL '#{set_url}' set in your options, "\
-              "so we're not replacing it with '#{options['url']}'!"
-          elsif not set_url
-            options['appium_lib.server_url'] = options['url']
+          # There are two ways to set the url and one has to be set
+          # appium_lib.server_url || url
+          # - disallow both being empty
+          # - disallow them being different in case both are set
+          # - otherwise, just take the one that is set and use it for both
+          server_url = options['appium_lib.server_url']
+          other_url = options['url']
+
+          if not option_set? server_url and not option_set? other_url and testdroid_testrun? options
```

Comment:
I'd either use `!` or `not` everywhere. Also, I would put brackets, e.g. `!option_set?(server_url)`. But these prefered ruby styles should probably defined once and project wide in the `.rubocop.yml` and is up to you :-)